src/datasources/balances-api/coingecko-api.service.ts

```diff
@@ -108,33 +116,49 @@ export class CoingeckoApi implements IPricesApi {
    * @param args.chainName Coingecko's name for the chain (see configuration)
    * @param args.tokenAddresses Array of token addresses which prices are being retrieved
    * @param args.fiatCode
-   * @returns Array of {@link AssetPrice}
+   * @returns Array of {@link CoingeckoAssetPrice}
    */
   async getTokenPrices(args: {
-    chainName: string;
+    chainId: string;
     tokenAddresses: string[];
     fiatCode: string;
-  }): Promise<AssetPrice[]> {
-    const pricesFromCache = await this._getTokenPricesFromCache(args);
-    const notCachedTokenPrices = difference(
-      args.tokenAddresses,
-      pricesFromCache.map((assetPrice) => Object.keys(assetPrice)).flat(),
-    );
-    const pricesFromNetwork = notCachedTokenPrices.length
-      ? await this._getTokenPricesFromNetwork({
-          ...args,
-          tokenAddresses: notCachedTokenPrices,
-        })
-      : [];
+  }): Promise<CoingeckoAssetPrice[]> {
+    try {
+      const lowerCaseFiatCode = args.fiatCode.toLowerCase();
+      const chainName = this.configurationService.getOrThrow<string>(
+        `prices.chains.${args.chainId}.chainName`,
+      );
+      const pricesFromCache = await this._getTokenPricesFromCache({
+        chainName,
+        tokenAddresses: args.tokenAddresses,
+        fiatCode: lowerCaseFiatCode,
+      });
+      const notCachedTokenPrices = difference(
+        args.tokenAddresses.map((address) => address.toLowerCase()),
+        pricesFromCache.map((assetPrice) => Object.keys(assetPrice)).flat(),
```

Comment:
To be doubly certain, I'd also convert these to lowercase.

Reply:
I've extracted it to `lowerCaseTokenAddresses` and used this variable to call `_getTokenPricesFromCache`, so all the addresses in both operands of `difference` will be lowercase. Let me know if you think this can be simplified further!

[a62b130](https://github.com/safe-global/safe-client-gateway/pull/1175/commits/a62b1305da4626e6b4362831001b84289acfed44)